### DISCUSSION TITLE
404 page is rendering HTML twice when using wrapPageElement

### DISCUSSION THREAD

**Discussion by Eytyy** [gatsbyjs/gatsby#35353] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

HTML inside the 404.js page is being duplicated when using Gatsby Browser API wrapPageElement. The issue is present with a fresh Gatsby setup.

![Screen Shot 2022-04-06 at 4 42 49 AM](https://user-images.githubusercontent.com/1340164/161878832-95e41130-813a-4d8a-b6a1-df61c5f03212.png)

### Reproduction Link

https://github.com/Eytyy/gatsby-404-duplicate-html

### Steps to Reproduce

1. wrap the element with random markup inside the wrapPageElement and export it from gatsby-browser.js
2. visit a page that doesn't exist
3. preview the 404 page

### Expected Result

HTML should be rendered once

### Actual Result

HTML is being rendered twice

### Environment

```shell
System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 100.0.4896.75
    Edge: 100.0.1185.29
    Firefox: 97.0.1
    Safari: 14.1.2
  npmPackages:
    gatsby: ^4.11.1 => 4.11.1
  npmGlobalPackages:
    gatsby-cli: 4.11.1
```

**Comment by LekoArts** [comment, marked as answer]
Hi!

You'll also need to implement the `wrapPageElement` API in `gatsby-ssr.js`: https://www.gatsbyjs.com/docs/reference/config-files/gatsby-ssr/#wrapPageElement

Otherwise the HTML and what the browser tries to re-hydrate is out of sync.

**Reply by mwskwong** [reply]
Nah, I did that and I still see the `Layout` still being wrapped in another `Layout` in the custom 404 page. It only appears in the custom 404 page and in DEV mode though.

**Comment by Eytyy** [comment]
Hey!

Thank you for the answer. Yes, this solved the issue, although I'm sure it was rendering duplicate HTML on the website I'm working on while having the API implemented in both `gatsby-browser` and `gatsby-ssr`, maybe I was staring at the development preview all this time thinking it was the production build 🙈  – it still renders the page twice on development mode, but I guess that's fine. Thank you again!

**Comment by creative-ae** [comment]
im having same issue. Layout rendering twice

**Comment by Vacilando** [comment]
Same issue - HTML rendered twice. In develop only if DEV_SSR is on, and on build.
